app/Console/Commands/PurgeAnonymousUsersCommand.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Console\Command;
+
+class PurgeAnonymousUsersCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'users:purge';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Does delete all user database entries, which are anonymous and had no activity for a long time';
+
+    public static $purgedBefore;
+
+    /**
+     * Alle anonymen nutzer, welche vor dem übergebenen datum ihre letzte aktivität hatten, werden gelöscht
+     * @param Carbon $date lösch timestamp. Muss in der vergangenheit liegen
+     */
+    public static function userPurgedBefore(Carbon $date)
+    {
+        if($date->isPast())
+            static::$purgedBefore = $date;
+    }
+
+    public static function getUserPurgedBeforeTime(){
+        return static::$purgedBefore?static::$purgedBefore:Carbon::now()->subMonth();
```

Comment:
Can be written shorter like this:
return static::$purgedBefore ?: Carbon::now()->subMonth();